<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>982</width>
    <height>692</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FlowCom Modbus Client</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <!-- Modbus Connection Group -->
    <item>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="portCombo"/>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Baud Rate:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QComboBox" name="baudRateCombo"/>
      </item>
      <item row="0" column="4">
       <widget class="QPushButton" name="connectButton">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <!-- Register Control Group -->
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Register Control - DEBUG &amp; TROUBLESHOOTING USE ONLY DO NOT USE</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Register:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="registerCombo"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Value:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="valueSpinBox">
         <property name="maximum">
          <number>65535</number>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="readButton">
           <property name="text">
            <string>Read</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="writeButton">
           <property name="text">
            <string>Write</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <!-- Value Display -->
    <item>
     <widget class="QLabel" name="valueLabel">
      <property name="text">
       <string>Value: --</string>
      </property>
     </widget>
    </item>
    <!-- Binary Display -->
    <item>
     <widget class="QTextEdit" name="binaryDisplay">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <!-- Auto Sequence Group -->
    <item>
     <widget class="QGroupBox" name="sequenceGroup">
      <property name="title">
       <string>Auto Sequence</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="startSequenceButton">
         <property name="text">
          <string>Start Sequence</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stopSequenceButton">
         <property name="text">
          <string>Stop Sequence</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <!-- Servo Port Connection Group (New) -->
    <item>
     <widget class="QGroupBox" name="servoPortGroup">
      <property name="title">
       <string>Servo Port Connection</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayoutServoPort">
       <item>
        <widget class="QLabel" name="labelServoPort">
         <property name="text">
          <string>Servo Port:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="servoPortCombo"/>
       </item>
       <item>
        <widget class="QPushButton" name="servoConnectButton">
         <property name="text">
          <string>Connect Servo</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <!-- Servo Controls Group -->
    <item>
     <widget class="QGroupBox" name="servoControlsGroup">
      <property name="title">
       <string>Servo Controls</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayoutServo">
       <item>
        <widget class="QPushButton" name="pwm1000Button">
         <property name="text">
          <string>PWM 1000</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pwm1500Button">
         <property name="text">
          <string>PWM 1500</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pwm2000Button">
         <property name="text">
          <string>PWM 2000</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="incrementButton">
         <property name="text">
          <string>Increase PWM by 1</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <!-- CSV Metadata Group -->
    <item>
     <widget class="QGroupBox" name="csvMetadataGroup">
      <property name="title">
       <string>CSV Metadata</string>
      </property>
      <layout class="QGridLayout" name="gridLayoutMetadata">
       <item row="0" column="0">
        <widget class="QLabel" name="labelFileName">
         <property name="text">
          <string>File Name:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="fileNameLineEdit"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="labelSerialNumber">
         <property name="text">
          <string>S/N:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="serialNumberLineEdit"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="labelCsvType">
         <property name="text">
          <string>CSV Type:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="csvTypeComboBox">
         <item>
          <property name="text">
           <string>Original</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Hand Modified</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
